src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -87,13 +87,28 @@ namespace AppInstaller::Logging
         m_isRuntimeEnabled = true;
     }
 
+    void TelemetryTraceLogger::SetUserSettingsStatus()
```

Comment:
Make this lazy init as well if there is an ordering problem with startup.

Reply:
AFAIU, UserSettings signleton is already a lazy init. Isn't it ? The problem is the first access of UserSettings is within the TelemetryTraceLogger constructor. Since UserSettings calls into TelemetryTraceLogger (this is NEW in the current PR) to write Diagnostics also to the logger with Activity ID (this is provided by TelemetryTraceLogger). This circular dependenc loop has created the hang. So I think the solution is to delay the first access of UserSettings by pulling it from out of TelemetryTraceLogger constructor.

Reply:
My understanding is UserSettings singleton is already a lazy init. Isn't it ? The problem is that the first access of UserSettings is within TelemetryTraceLogger constructor. And since UserSettings logs for no json file case as Diagnostic event with Activity Id (Activity Id in Diagnostic trace is new in this PR and it's access is via a method in TelemetryTraceLogger). This causes a circular dependency in runtime and hangs due to the call_once mutex around TelemetryTraceLogger's lazy init.

I think the solution is to move out the UserSettings's first access from TelemetryTraceLogger constructor to immediately after the construction.

Reply:
I didn't mean UserSettings, I meant the bool.  However, if there is a circular dependency there will always be one.  Right now you have avoided it by not having `SetUserSettingsStatus` write to the bool in an exclusive way, otherwise you would be right back to the same situation.

I honestly don't know how it works right now, given that the same thing will happen but using the implicit static init_once.

The best way to solve this that is coming to me is to make the bool lazy init, but in a way that reentrant reads on the bool will return the default value until it is set.  Given that it is coming from a setting, that makes sense (if there is an error reading settings we can't really get a non-default value).

What you have now is the simplest way to achieve that, but it requires that the caller remember to do the extra step after construction, which is *always* required.

I'm suggesting something far more complicated I agree, but with the goal of preventing bugs in the future.

Reply:
Ok, I think that I have a solution that is half way between that doesn't make things (as) complicated: Have the Globals actually create the Telemetry object and call `SetUserSettingsStatus` itself.
```
void Globals::SetCurrent()
{
  m_telemetry = std::make_unique<Telemetry>();
  tl_currentGlobals = this;
  m_telemetry->SetUserSettingsStatus();
}
```
Telemetry is no longer under the call_once, and there is only one place where we have to do this secondary initialization call.